catalog/core/catalog-core-api-impl/src/main/java/ddf/catalog/data/impl/MetacardImpl.java

```diff
@@ -759,6 +762,8 @@ private void writeObject(ObjectOutputStream stream) throws IOException {
             stream.writeObject(mt);
         }
 
+        stream.writeObject(getSourceId());
```

Comment:
I am fairly confident this will break backwards compatibility with serialized metacards.  I do not think this should be included in this fix.

It was intentional to not include the source id when metacards are exported and imported.  The system that it is imported into will set its source id.